From 2018cad7a4607b51b699b498611f9dc93aaee869 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:28:38 -0400
Subject: [acpi] libata: Remove ata_dev_acpi_handle stub in libata.h

Message-id: <1411129776-21656-4-git-send-email-prarit@redhat.com>
Patchwork-id: 96360
O-Subject: [RHEL7.1 PATCH BZ 1141846 03/61] libata: acpi: Remove ata_dev_acpi_handle stub in libata.h
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 03f7b802b876904971f23f00f562061309e95e08
Author: Aaron Lu <aaron.lu@intel.com>
Date:   Tue Aug 27 14:07:12 2013 -0400

    libata: acpi: Remove ata_dev_acpi_handle stub in libata.h

    The ata_dev_acpi_handle is defined in libata-acpi.c and the only
    external user is libata-zpodd.c, which is built when CONFIG_ATA_ACPI
    is set, so there is no need to make an empty stub function for
    ONFIG_ATA_ACPI case in libata.h. It also causes compile errors due to
    acpi_handle is not defined when !CONFIG_ACPI. This patch fixes this
    problem by removing the empty stub.

    Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
    Signed-off-by: Aaron Lu <aaron.lu@intel.com>
    Signed-off-by: Tejun Heo <tj@kernel.org>

Cc: David Milburn <dmilburn@redhat.com>
Cc: Jeff Garzik <jgarzik@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/ata/libata.h b/drivers/ata/libata.h
index f9d22a3..91a5eb9 100644
--- a/drivers/ata/libata.h
+++ b/drivers/ata/libata.h
@@ -133,7 +133,6 @@ static inline void ata_acpi_set_state(struct ata_port *ap,
           pm_message_t state) { }
 static inline void ata_acpi_bind_port(struct ata_port *ap) {}
 static inline void ata_acpi_bind_dev(struct ata_device *dev) {}
-static inline acpi_handle ata_dev_acpi_handle(struct ata_device *dev) { return NULL; }
 static inline void ata_acpi_bind(struct ata_device *dev) { }
 static inline void ata_acpi_unbind(struct ata_device *dev) { }
 #endif
-- 
1.7.1